sim-lib/src/lib.rs

```diff
@@ -383,8 +418,8 @@ impl Simulation {
         for (id, node) in self.nodes.iter().filter(|(pk, _)| {
             self.activity
                 .iter()
-                .map(|a| a.source)
-                .collect::<HashSet<PublicKey>>()
+                .map(|a| a.source.get_pk().unwrap())
```

Comment:
okay but can we avoid the assumption of having called `validate_activity` before this point?
Same sentiment for the other `get_pk.unwrap()` points

Reply:
Can we? Sure, but I don't think we should. `validate_activity` is our sanity checker, we could modify all these calls to account for an alias being here, but that's exactly what is not supposed to happen. If any `NodeId` contains an alias at this point, it means the code if wrong (hence why there was a "report message".

I could patch the code so it does not unwrap and it fails more "elegantly", gut all in all we want it to fail if that happens

Reply:
Alternatively, we could have two different classes to account for node definition and activity definition, one that interfaces with the config file, and where we allow aliases and pks (data is parsed as `NodeId`), and an internal one that gets constructed from the former, and only accepts pks. In that way, we can be sure at compile time that everything is identified by pk internally.

Reply:
I've split this into two sets of data structures in [2546166](https://github.com/bitcoin-dev-project/sim-ln/pull/103/commits/2546166018c9dbcaf51b6995b099b315e4070919). The `Parsers` that interact directly with the config file and allow `NodeId`s to be loaded, and the regular ones which roll back to use `PublicKey`s and can be built from the former.

This commit can be squashed with the previous one. Also, I'm happy to replace `Parser` with any other better term we can come up with (`Loader`, `Reader`, ...)